pkg/volume/flexvolume/probe_test.go

```diff
@@ -0,0 +1,294 @@
+	assert.Equal(t, 0, len(plugins))
+	assert.NoError(t, err)
+}
+
+// Probes newly added drivers after prober is running.
+func TestProberAddDriver(t *testing.T) {
+	// Arrange
+	_, fs, watcher, prober := initTestEnvironment(t)
+	prober.Probe()
+	updated, _, _ := prober.Probe()
+	assert.False(t, updated)
+
+	// Call probe after a file is added. Should return true.
+
+	// Arrange
+	const driverName2 = "fake-driver2"
+	driverPath := path.Join(pluginDir, driverName2)
+	installDriver(driverName2, fs)
+	watcher.TriggerEvent(fsnotify.Create, driverPath)
+	watcher.TriggerEvent(fsnotify.Create, path.Join(driverPath, driverName2))
+
+	// Act
+	updated, plugins, err := prober.Probe()
+
+	// Assert
+	assert.True(t, updated)
+	assert.Equal(t, 2, len(plugins))                                     // 1 existing, 1 newly added
+	assert.Equal(t, driverPath, watcher.watches[len(watcher.watches)-1]) // Checks most recent watch
+	assert.NoError(t, err)
+
+	// Call probe again, should return false.
+
+	// Act
+	updated, _, err = prober.Probe()
+	// Assert
+	assert.False(t, updated)
+	assert.NoError(t, err)
+
+	// Call probe after a non-driver file is added in a subdirectory. Should return true.
+
+	// Arrange
+	fp := path.Join(driverPath, "dummyfile")
+	fs.Create(fp)
+	watcher.TriggerEvent(fsnotify.Create, fp)
+
+	// Act
+	updated, plugins, err = prober.Probe()
+
+	// Assert
+	assert.True(t, updated)
+	assert.Equal(t, 2, len(plugins)) // Number of plugins should not change.
+	assert.NoError(t, err)
+
+	// Call probe again, should return false.
+	// Act
+	updated, _, err = prober.Probe()
+	// Assert
+	assert.False(t, updated)
+	assert.NoError(t, err)
+}
+
+func TestEmptyPluginDir(t *testing.T) {
+	// Arrange
+	fs := utilfs.NewFakeFs()
+	watcher := NewMockWatcher()
+	prober := &flexVolumeProber{
+		pluginDir: pluginDir,
+		watcher:   watcher,
+		fs:        fs,
+		factory:   mockPluginFactory{error: false},
+	}
+	prober.Init()
+
+	// Act
+	updated, plugins, err := prober.Probe()
+
+	// Assert
+	assert.True(t, updated)
+	assert.Equal(t, 0, len(plugins))
+	assert.NoError(t, err)
+}
+
+// Issue an event to remove plugindir. New directory should still be watched.
+func TestRemovePluginDir(t *testing.T) {
+	// Arrange
+	driverPath, fs, watcher, _ := initTestEnvironment(t)
+	fs.RemoveAll(pluginDir)
+	watcher.TriggerEvent(fsnotify.Remove, path.Join(driverPath, driverName))
+	watcher.TriggerEvent(fsnotify.Remove, driverPath)
+	watcher.TriggerEvent(fsnotify.Remove, pluginDir)
+
+	// Act: The handler triggered by the above events should have already handled the event appropriately.
+
+	// Assert
+	assert.Equal(t, 3, len(watcher.watches)) // 2 from initial setup, 1 from new watch.
+	assert.Equal(t, pluginDir, watcher.watches[len(watcher.watches)-1])
+}
+
+// Issue multiple events and probe multiple times. Should give true, false, false...
+func TestProberMultipleEvents(t *testing.T) {
+	// Arrange
+	_, fs, watcher, prober := initTestEnvironment(t)
+	for i := 0; i < 5; i++ {
```

Comment:
put the number of iterations in a variable so you can adjust more easily/calculate the number for your assertions w/o hardcoding

Reply:
ACK